CommonCompilerConfig.cmake

```diff
@@ -61,6 +61,13 @@
 #                      and shared libraries (DLLs) with optimizations that are
 #                      compatible with INTERNAL_CXX_FLAGS_OPTIMIZED
 #
+# COMMON_STATICLIB_CMAKE_DEFINITIONS_OPTIMIZED: list of DMAKE_AR and DMAKE_RANLIB flags required for gcc >= 4.9
```

Comment:
The name of the variable is more generic than the provided description. Perhaps document it as: list of -D cmake definitions corresponding to `EXTERNAL_C_FLAGS_OPTIMIZED`. If you think it's useful, you may then site those details as examples.

Ditto for `COMMON_STATICLIB_CONFIGURE_DEFINITIONS_OPTIMIZED_STR`.

Also, since these are used only for building external projects due to the cmake bug https://gitlab.kitware.com/cmake/cmake/issues/15547, let's change the prefix of both of these variables from `COMMON_` to `EXTERNAL_` for consistency with other such flags here (sorry for suggesting otherwise earlier).